src/libasr/codegen/llvm_utils.cpp

```diff
@@ -2344,7 +2344,17 @@ namespace LCompilers {
         llvm::Type* el_type = std::get<2>(typecode2listtype[type_code]);
         resize_if_needed(list, current_end_point, current_capacity,
                          type_size, el_type, module);
-
+        llvm::Value* adjusted_pos = builder->CreateSelect(
+            builder->CreateICmpSGT(pos, current_end_point),
+            current_end_point,
+            pos
+        );
+        adjusted_pos = builder->CreateSelect(
+            builder->CreateICmpSGE(adjusted_pos, llvm::ConstantInt::get(context, llvm::APInt(32, 0))),
+            adjusted_pos,
+            llvm::ConstantInt::get(context, llvm::APInt(32, 0))
+        );
+        
```

Comment:
Seems like an issue with edge case. Note that `current_end_point` equals `len(list)` all the times.

https://github.com/lcompilers/lpython/blob/57ebbc81e23854d3af50c2d5042f2f1b184941b5/src/libasr/codegen/llvm_utils.cpp#L2252-L2254